Documentation/sources/exd-cbe-beta/apis-usage.md

```diff
@@ -0,0 +1,234 @@
+# APIs Usage
+
+This document details the Messaging SDK APIs that can be used to implement code-based experiences in mobile apps.
+
+## Code-based experiences APIs
+
+- [updatePropositionsForSurfaces](#updatePropositionsForSurfaces)
+- [getPropositionsForSurfaces](#getPropositionsForSurfaces)
+
+---
+
+### updatePropositionsForSurfaces
+
+Dispatches an event for the Edge network extension to fetch personalization decisions from the AJO campaigns for the provided surfaces array. The returned decision propositions are cached in-memory by the Messaging extension.
+
+To retrieve previously cached decision propositions, use `getPropositionsForSurfaces` API.
+
+#### Java
+
+##### Syntax
+
+```java
+public static void updatePropositionsForSurfaces(@NonNull final List<Surface> surfaces)
+```
+
+##### Example
+
+```java
+final Surface surface1 = new Surface("myActivity#button");
+final Surface surface2 = new Surface("myActivityAttributes");
+
+final List<Surface> surfaces = new ArrayList<>();
+surfaces.add(surface1);
+surfaces.add(surface2);
+
+Messaging.updatePropositionsForSurfaces(surfaces)
+```
+
+---
+
+### getPropositionsForSurfaces
+
+Retrieves the previously fetched propositions from the SDK's in-memory propositions cache for the provided surfaces. The completion handler is invoked with the decision propositions corresponding to the given surfaces or AEPError, if it occurs. 
```

Comment:
Should "completion handler" be changed to "Adobe Callback" for the android documentation? In addition, `AEPError` is `AdobeError` on the android side.